### PR TITLE
only parse TMS when used

### PR DESCRIPTION
**Pull request by vincentsarago**
ref https://github.com/AndrewAnnex/planetcantile/pull/2#discussion_r1046879136

```
## Main
time python -c'import morecantile'
python -c'import morecantile'  0.19s user 0.03s system 98% cpu 0.222 total

## This Branch
time python -c'import morecantile'
python -c'import morecantile'  0.10s user 0.02s system 96% cpu 0.132 total
```

### CODE REVIEW

File: morecantile/defaults.py [hunk 1]
```diff
@@ -33,7 +33,12 @@ def get(self, identifier: str) -> TileMatrixSet:
         if identifier not in self.tms:
             raise InvalidIdentifier(f"Invalid identifier: {identifier}")
 
-        return self.tms[identifier]
+        tms = self.tms[identifier]
+        if isinstance(tms, pathlib.Path):
+            tms = TileMatrixSet.parse_file(tms)
+            self.tms[identifier] = tms
```
Comment (vincentsarago) [inline morecantile/defaults.py:39]: if TMS is a `Path` object we parse it and store it in the dict so future call won't need to `parse`

**Comment by AndrewAnnex** [comment]
I think this great as is! Simple change with no downsides I can see, and I can imagine those milliseconds adding up...

**Comment by vincentsarago** [comment]
@geospatial-jeff any comments?

**Review by geospatial-jeff — APPROVED** [review]
LGTM!

**Pull request merged**